### PR TITLE
integration: order and status checks, no pending server messages

### PR DESCRIPTION
**Pull request by chappjc**
This includes:
- removal of server-side message caching (pending messages that were stored until user reconnects)
- 'connect' response including user's active orders
- 'order_status' server route and client support for checking orders as needed
- 'connect' response including user's active matches
- 'match_status' server route and client support for checking matches as needed

Matches status is processed before order statuses.

`status-integ` PRs: https://github.com/decred/dcrdex/pull/677, https://github.com/decred/dcrdex/pull/699, https://github.com/decred/dcrdex/pull/687, https://github.com/decred/dcrdex/pull/704, https://github.com/decred/dcrdex/pull/714

Note: this PR will be merged via **rebase, not squash** so each of the commits are replayed on master.

### CODE REVIEW

**Comment by chappjc** [comment]
This PR simply needs testing, and notes for breaking changes (API, db scheme, etc.).  The individual commits have already gone through review.

File: client/core/core.go [hunk 1]
```diff
@@ -2519,45 +2610,79 @@ func (c *Core) authDEX(dc *dexConnection) error {
...
 
 		// Send a "Match resolution error" if there are extra match messages.
 		if len(extras) > 0 {
-			details := fmt.Sprintf("%d matches reported by %s were not found for %s.",
-				len(extras), dc.acct.host, trade.token())
-			c.notify(newOrderNote("Match resolution error", details, db.ErrorLevel, corder))
-			// t.negotiate(extras) // missed match message request, but match not revoked (?!)
 			for _, extra := range extras {
-				log.Errorf("DEX %s reported match %s which is not a known active match for order %s",
+				log.Debug("DEX %s reported match %s which is not a known active match for order %s",
```
Comment (buck54321) [inline client/core/core.go:2636]: Need a `Debugf` here, not `Debug`.

**Review by buck54321 — COMMENTED** [review]
Looks and tests well. Other API used only internally, my quick tally of breaking changes is 

1. `IsRevoked` field for `db.MatchProof` is now a method that returns `proof.ServerRevoked || proof.SelfRevoked`. (actually, this one isn't used outside **core** probably)
2. `msgjson.OrderStatusResult` renamed to `OrderStatus` and `Filled` field removed.
3. `msgjson.ConnectResult` `Matches` field renamed to `ActiveMatches` (json `matches` -> `activematches`).

**Comment by chappjc** [comment]
Thanks.  Debugf fix rebased into this `match_status resolution` commit via fixup.

**Pull request merged**